tests/dir_helpers.py

```diff
@@ -156,7 +156,18 @@ def dvc_add(self, filenames, commit=None):
         stages = self.dvc.add(filenames)
         if commit:
             stage_paths = [s.path for s in stages]
-            self.scm_add(stage_paths, commit=commit)
+
+            def to_gitignore(stage_path):
+                from dvc.scm import Git
+
+                return os.path.join(os.path.dirname(stage_path), Git.GITIGNORE)
+
+            gitignores = [
+                to_gitignore(s)
+                for s in stage_paths
+                if os.path.exists(to_gitignore(s))
+            ]
```

Comment:
I wonder if all you need to do is:
```python
paths = self.scm.files_to_track
```

Reply:
The problem here is that `scm_context` wrapper resets the `files_to_track` after successful `add`.